### DISCUSSION TITLE
Why does the singularity container not have GPU?

### DISCUSSION THREAD

**Discussion by brando90** [princeton-vl/CoqGym#22] [Q&A] [answered]
I was thinking that self contained container where everything can be done where it fully works and is installed/used once would be the best. But I found out that the container given does not work with GPU. I am curious, why does it not work with GPU and is it hard to extend it to work with GPU?

(also testing the first discussion feature!)

**Comment by brando90** [comment]
e.g. it could help avoid this sort of error: https://github.com/princeton-vl/CoqGym/issues/20

**Comment by yangky11** [comment, marked as answer]
That was because Singularity didn't have reliable GPU support at the time when CoqGym was released. But it looks like things are different now: https://sylabs.io/guides/3.5/user-guide/gpu.html. 

I plan to add GPU support to CoqGym's Singularity container when I get a chance, though it may take some time.
